src/adapter/src/coord/message_handler.rs

```diff
@@ -676,4 +684,101 @@ impl<S: Append + 'static> Coordinator<S> {
             });
         }
     }
+
+    #[tracing::instrument(level = "debug", skip_all)]
+    /// Finishes sequencing a command that was waiting on a real time recency timestamp.
+    async fn message_real_time_recency_timestamp(
+        &mut self,
+        conn_id: ConnectionId,
+        real_time_recency_ts: mz_repr::Timestamp,
+    ) {
+        if let Some(real_time_recency_context) =
+            self.pending_real_time_recency_timestamp.remove(&conn_id)
+        {
+            match real_time_recency_context {
+                RealTimeRecencyContext::ExplainTimestamp {
```

Comment:
Yesterday, I thought that we wouldn't need to re-validate explain timestamp, but I realize that I was wrong, I'll fix this today.

Reply:
Done